wrf/ts.py

```diff
@@ -0,0 +1,199 @@
+                          for prefix in self.tslist.index ]
+        for prefix,fpath in zip(self.tslist.index, self.filelist):
+            if os.path.isfile(fpath) and not overwrite:
+                if self.verbose: print('Reading',fpath)
+                self.data[prefix] = xr.open_dataset(fpath)
+            else:
+                if self.verbose: print('Creating',fpath)
+                self.data[prefix] = self._process_tower(prefix,
+                                                        heights=heights,
+                                                        height_var=height_var,
+                                                        approx_height=approx_height,
+                                                        outfile=fpath)
+
+    def _process_tower(self,prefix,
+                       heights=None,height_var=None,approx_height=True,
+                       outfile=None):
+        """Use Tower.to_dataframe() to create a dataframe, to which we
+        add tower latitude/longitude. Setting them as indices makes the
+        recognizable as coordinates by xarray.
+        """
+        towerfile = '{:s}.d{:02d}.*'.format(prefix, self.domain)
+        fpath = os.path.join(self.towerdir,towerfile)
+        # create Tower object
+        tow = Tower(fpath,varlist=self.varnames)
+        # set up height variable if needed
+        if heights is not None:
+            assert (height_var is not None), 'height attribute unknown'
+        if height_var == 'ph':
+            # this creates a time-heightlevel varying height
+            tow.height = getattr(tow,height_var) - tow.stationz
+            if approx_height:
+                mean_height = np.mean(tow.height, axis=0)
+                if self.verbose:
+                    # diagnostics
+                    zmax = np.max(heights)
+                    heights_within_micro_dom = np.ma.masked_array(
+                            tow.height, tow.height > zmax)
+                    stdev0 = np.std(tow.height, axis=0)
+                    kmax0 = np.argmax(stdev0)
+                    stdev = np.std(heights_within_micro_dom, axis=0)
+                    kmax = np.argmax(stdev)
+                    print('max stdev in height at (z~={:g}m) : {:g}'.format(
+                            mean_height[kmax0], stdev0[kmax0]))
+                    print('max stdev in height (up to z={:g} m) at (z~={:g} m) : {:g}'.format(
+                            np.max(heights_within_micro_dom), mean_height[kmax], stdev[kmax]))
+                tow.height[:,:] = mean_height[np.newaxis,:] 
+                for k,hgt in enumerate(mean_height):
+                    assert np.all(tow.height[:,k] == hgt)
+        elif height_var != 'height':
+            raise ValueError('Unexpected height_var='+height_var+'; heights not calculated')
+        # now convert to a dataframe (note that height interpolation
+        # will be (optionally) performed here
+        df = tow.to_dataframe(start_time=self.starttime,
```

Comment:
@phawbeck This is how I'm calling `Tower.to_dataframe()`... should be pretty straightforward 🤞